apps/meteor/client/views/modal/uikit/hooks/useValues.ts

```diff
@@ -0,0 +1,48 @@
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import type { LayoutBlock } from '@rocket.chat/ui-kit';
+import { useReducer } from 'react';
+
+type LayoutBlockWithElement = Extract<LayoutBlock, { element: unknown }>;
+type LayoutBlockWithElements = Extract<LayoutBlock, { elements: readonly unknown[] }>;
+type ElementFromLayoutBlock = LayoutBlockWithElement['element'] | LayoutBlockWithElements['elements'][number];
+
+const hasElementInBlock = (block: LayoutBlock): block is LayoutBlockWithElement => 'element' in block;
+const hasElementsInBlock = (block: LayoutBlock): block is LayoutBlockWithElements => 'elements' in block;
+const hasInitialValueAndActionId = (
+	element: ElementFromLayoutBlock,
+): element is ElementFromLayoutBlock & { initialValue: unknown; actionId: string } =>
```

Comment:
`nitpick` Why not `Extract<ElementFromLayoutBlock, { initialValue: unknown; actionId: string }>`? It's almost the same, but it handles type names in unions (like `ElementFromLayoutBlock`) a little bit better.